modules/graphics/src/renderable.cxx

```diff
@@ -0,0 +1,15 @@
+#include "moon/renderable.hxx"
+
+namespace Moon {
+  Renderable::Renderable() {
+    m_shader = NULL;
+  }
+
+  void Renderable::SetShader(Moon::Shader *shader) {
+    m_shader = shader;
+  }
+
+  Moon::Shader* Renderable::GetShader() {
+    return m_shader;
```

Comment:
m_shader is private, which is why you can't set it, from the child class, blame C++.